src/Hintman/Hint/NoNewlineAtFileEnd.hs

```diff
@@ -0,0 +1,67 @@
+module Hintman.Hint.NoNewlineAtFileEnd
+       ( getNoNewlineAtFileEndComments
+       ) where
+
+import Hintman.Core.Hint (Hint, HintType (NoNewlineAtFileEnd), Line (..), simpleSuggestion, toLines)
+import Hintman.Core.PrInfo (ModifiedFile (..))
+import Hintman.Core.Review (Comment (..), createComment)
+
+import qualified Data.Text as T
+
+
+-- | Creates 'Comment's on adding newlines in all modified files.
+getNoNewlineAtFileEndComments :: [ModifiedFile] -> [Comment]
+getNoNewlineAtFileEndComments = mapMaybe getFileNoNewlineAtFileEndComment
+
+{- | Takes a file and returns a comment if file doesn't end with a '\n'.
+The suggestion text for the line in comment is the context of the line with '\n' appended. -}
+getFileNoNewlineAtFileEndComment :: ModifiedFile -> Maybe Comment
+getFileNoNewlineAtFileEndComment mf@ModifiedFile{..} = spawnComment $
+    maybeGetNonEmptyLastLine fileLines diffText
+  where
+    diffText :: Text
+    diffText = decodeUtf8 mfContent
+
+    fileLines :: [Line]
+    fileLines = toLines diffText
+
+    {- | Spawns a "NoNewlineAtFileEnd" comment on a given line, suggesting the same
+    line, but with '\n' appended. -}
+    spawnComment :: Maybe Line -> Maybe Comment
+    spawnComment = (=<<) (\line -> createComment mf line $ createHint line)
```

Comment:
Instead of creating a function from `Maybe a` to `Maybe b` you can create a function from `Line` to `Maybe Comment` writing it like this:

```haskell
spawnComment line = ...
```

And then instead of `spawnComment $ ...` you can just `spawnComment =<< ...`. It will make code structurally a bit simpler to understand.